File: src/main/java/com/itmo/microservices/demo/payment/externalPaymentService/client/ExternalServiceClient.java
package com.itmo.microservices.demo.payment.externalPaymentService.client;

import com.itmo.microservices.demo.payment.externalPaymentService.client.model.ExternalServiceRequest;
import com.itmo.microservices.demo.payment.externalPaymentService.client.model.ExternalServiceResponse;

public interface ExternalServiceClient {

    ExternalServiceResponse executePayment(ExternalServiceRequest request);
}

File: src/main/kotlin/com/itmo/microservices/demo/delivery/externalDeliveryService/client/ExternalServiceClient.kt
package com.itmo.microservices.demo.delivery.externalDeliveryService.client

import com.itmo.microservices.demo.delivery.externalDeliveryService.client.model.ExternalServiceRequest
import com.itmo.microservices.demo.delivery.externalDeliveryService.client.model.ExternalServiceResponse

interface ExternalServiceClient {
    fun getDeliveryTransactionId(request: ExternalServiceRequest): ExternalServiceResponse?
    fun getDeliveryParams(request: ExternalServiceRequest, transactionId: String): ExternalServiceResponse?
}